### PR TITLE
Run Jupyter notebooks in CI

### PR DESCRIPTION
**Pull request by sritchie**
This PR:

- adjusts our `ci.yml` job to run our tests and our notebooks in parallel
- updates all notebooks to compatibility with `nightly` so they build and pass tests
- deletes failing notebooks, so we can add them via a new PR

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,11 +10,8 @@ on:
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
-env:
-  CODECOV_TOKEN: 1abb2fd7-0cb5-433c-b2cc-1596466aeb56
-
 jobs:
-  test:
```
Comment (sritchie) [inline .github/workflows/ci.yml:17]: I haven't done a combo yml file like this, but I like sharing the setup... let's see what happens!

File: noxfile.py [hunk 2]
```diff
@@ -51,8 +51,6 @@ def tests(session):
         "notebooks",
         "--ignore",
         "benchmarks",
-        "--ignore",
```
Comment (sritchie) [inline noxfile.py:54]: duplicate pair of lines

File: noxfile.py [hunk 3]
```diff
@@ -126,7 +139,8 @@ def mypy(session) -> None:
     session.run_always("poetry", "install", "--with", "dev", external=True)
     session.run("mypy", *args)
     if not session.posargs:
-        session.run("mypy", f"--python-executable={sys.executable}", "noxfile.py")
```
Comment (sritchie) [inline noxfile.py]: This may need to be `poetry run mypy`...

**Comment by sritchie** [comment]
Here's our final list of errors from the notebooks. @femtomc , we'd love your help figuring out which of these we should treat by deletion and which merit proper fixes.

```python
....FFF...F.                                                                                                                         [100%]
================================================================= FAILURES ==================================================================
/Users/sritchie/code/python/genjax/notebooks/concepts/combinators/intro_to_combinators.ipynb ________________________
[gw6] darwin -- Python 3.10.9 /Users/sritchie/code/python/genjax/.venv/bin/python
---------------------------------------------------------------------------
tr.indices
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[6], line 1
----> 1 tr.indices

AttributeError: 'UnfoldTrace' object has no attribute 'indices'
__________________________________ /Users/sritchie/code/python/genjax/notebooks/advanced/gensp/gensp.ipynb __________________________________
[gw0] darwin -- Python 3.10.9 /Users/sritchie/code/python/genjax/.venv/bin/python
---------------------------------------------------------------------------
gensp_model = gensp.chm_dist(model, selection=genjax.select("x", "y"), custom_q=None)
gensp_model
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[6], line 1
----> 1 gensp_model = gensp.chm_dist(model, selection=genjax.select("x", "y"), custom_q=None)
      2 gensp_model

AttributeError: module 'genjax.gensp' has no attribute 'chm_dist'
____________________ /Users/sritchie/code/python/genjax/notebooks/concepts/gradient_interfaces/gradient_interfaces.ipynb ____________________
[gw5] darwin -- Python 3.10.9 /Users/sritchie/code/python/genjax/.venv/bin/python
---------------------------------------------------------------------------
def viz(ax, x, y, **kwargs):
    sns.scatterplot(x=x, y=y, ax=ax, **kwargs)


f, axes = plt.subplots(3, 3, figsize=(9, 9), sharex=True, sharey=True, dpi=280)
jitted = jax.jit(model.simulate)
trs = []
for ax in axes.flatten():
    key, sub_key = jax.random.split(key)
    tr = jitted(sub_key, ())
    x = tr["x", "pos"]
    y = tr["y", "pos"]
    trs.append(tr)
    viz(ax, x, y, marker=".")

plt.show()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[4], line 10
      8 for ax in axes.flatten():
      9     key, sub_key = jax.random.split(key)
---> 10     tr = jitted(sub_key, ())
     11     x = tr["x", "pos"]
     12     y = tr["y", "pos"]

    [... skipping hidden 4 frame]

File ~/code/python/genjax/src/genjax/_src/core/pytree/pytree.py:139, in Pytree.unflatten(cls, data, xs)
    106 @classmethod
    107 def unflatten(cls, data, xs):
    108     """Given an implementation of `flatten` (c.f. above), `unflatten` is
    109     automatically defined and registered with JAX's `Pytree` system.
    110
   (...)
    137         ```
    138     """
--> 139     return cls(*data, *xs)

TypeError: RepeatTrace.__init__() missing 1 required positional argument: 'args'
___________________ /Users/sritchie/code/python/genjax/notebooks/intermediate/gaussian_processes/gaussian_processes.ipynb ___________________
[gw0] darwin -- Python 3.10.9 /Users/sritchie/code/python/genjax/.venv/bin/python
---------------------------------------------------------------------------
def viz(ax, x, y, **kwargs):
    ax.plot(x, y, **kwargs)


f, axes = plt.subplots(3, 3, figsize=(9, 9), sharex=True, sharey=True, dpi=280)
jitted = jax.jit(model.simulate)

for ax in axes.flatten():
    key, sub_key = jax.random.split(key)
    tr = jitted(sub_key, (x,))
    y = tr.get_retval()
    viz(ax, x, y, marker=".")

plt.show()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[10], line 10
      8 for ax in axes.flatten():
      9     key, sub_key = jax.random.split(key)
---> 10     tr = jitted(sub_key, (x,))
     11     y = tr.get_retval()
     12     viz(ax, x, y, marker=".")

    [... skipping hidden 12 frame]

File <@beartype(genjax._src.generative_functions.static.static_gen_fn.StaticGenerativeFunction.simulate) at 0x126846710>:53, in simulate(__beartype_func, __beartype_conf, __beartype_get_violation, __beartype_object_4898304960, __beartype_object_5184426576, *args, **kwargs)

File ~/code/python/genjax/src/genjax/_src/generative_functions/static/static_gen_fn.py:136, in StaticGenerativeFunction.simulate(self, key, args)
    127 @typecheck
    128 def simulate(
    129     self,
    130     key: PRNGKey,
    131     args: Tuple,
    132 ) -> StaticTrace:
    133     syntax_sugar_handled = push_trace_overload_stack(
    134         handler_trace_with_static, self.source
    135     )
--> 136     (args, retval, address_choices, score), cache_state = simulate_transform(
    137         syntax_sugar_handled
    138     )(key, args)
    139     return StaticTrace(
    140         self,
    141         args,
   (...)
    145         score,
    146     )

File ~/code/python/genjax/src/genjax/_src/generative_functions/static/static_transforms.py:307, in simulate_transform.<locals>.wrapper(key, args)
    304 @functools.wraps(source_fn)
    305 def wrapper(key, args):
    306     stateful_handler = SimulateHandler(key)
--> 307     retval = forward(source_fn)(stateful_handler, *args)
    308     (
    309         address_choices,
    310         cache_state,
    311         score,
    312     ) = stateful_handler.yield_state()
    313     return (
    314         args,
    315         retval,
    316         address_choices,
    317         score,
    318     ), cache_state

File <@beartype(genjax._src.core.interpreters.forward.forward.wrapped) at 0x2927355a0>:31, in wrapped(__beartype_func, __beartype_conf, __beartype_get_violation, __beartype_object_4833370672, *args, **kwargs)

File ~/code/python/genjax/src/genjax/_src/core/interpreters/forward.py:254, in forward.<locals>.wrapped(stateful_handler, *args)
    250 @functools.wraps(f)
    251 @typecheck
    252 def wrapped(stateful_handler: StatefulHandler, *args):
    253     interpreter = ForwardInterpreter()
--> 254     return interpreter.run_interpreter(
    255         stateful_handler,
    256         f,
    257         *args,
    258     )

File ~/code/python/genjax/src/genjax/_src/core/interpreters/forward.py:237, in ForwardInterpreter.run_interpreter(self, stateful_handler, fn, *args, **kwargs)
    234 def _inner(*args):
    235     return fn(*args, **kwargs)
--> 237 _closed_jaxpr, (flat_args, _, out_tree) = stage(_inner)(*args)
    238 _jaxpr, consts = _closed_jaxpr.jaxpr, _closed_jaxpr.literals
    239 flat_out = self._eval_jaxpr_forward(
    240     stateful_handler,
    241     _jaxpr,
    242     consts,
    243     flat_args,
    244 )

File ~/code/python/genjax/src/genjax/_src/core/interpreters/staging.py:76, in stage.<locals>.wrapped(*args, **kwargs)
     74 flat_fun, out_tree = api_util.flatten_fun_nokwargs(fun, in_tree)
     75 flat_avals = safe_map(get_shaped_aval, flat_args)
---> 76 typed_jaxpr = cached_stage_dynamic(flat_fun, tuple(flat_avals))
     77 return typed_jaxpr, (flat_args, in_tree, out_tree)

    [... skipping hidden 1 frame]

File ~/code/python/genjax/src/genjax/_src/core/interpreters/staging.py:63, in cached_stage_dynamic(flat_fun, in_avals)
     61 @lu.cache
     62 def cached_stage_dynamic(flat_fun, in_avals):
---> 63     jaxpr, _, consts = pe.trace_to_jaxpr_dynamic(flat_fun, in_avals)
     64     typed_jaxpr = jax_core.ClosedJaxpr(jaxpr, consts)
     65     return typed_jaxpr

    [... skipping hidden 4 frame]

File ~/code/python/genjax/src/genjax/_src/core/interpreters/forward.py:235, in ForwardInterpreter.run_interpreter.<locals>._inner(*args)
    234 def _inner(*args):
--> 235     return fn(*args, **kwargs)

File ~/code/python/genjax/src/genjax/_src/generative_functions/supports_callees.py:52, in push_trace_overload_stack.<locals>.wrapped(*args)
     50 def wrapped(*args):
     51     GLOBAL_TRACE_HANDLER_STACK.append(handler)
---> 52     ret = fn(*args)
     53     GLOBAL_TRACE_HANDLER_STACK.pop()
     54     return ret

Cell In[9], line 7, in model(data)
      5 kernel_scaled = kernels.Cosine(scale=scale)
      6 gp = ggp.GaussianProcess(kernel_scaled)
----> 7 y = trace("y", gp)(data)
      8 return y

File ~/code/python/genjax/src/genjax/_src/generative_functions/static/static_transforms.py:149, in trace.<locals>.<lambda>(*args)
    138 """Invoke a generative function, binding its generative semantics with the
    139 current caller.
    140
   (...)
    146     callable: A callable which wraps the `trace_p` primitive, accepting arguments (`args`) and binding the primitive with them. This raises the primitive to be handled by `StaticGenerativeFunction` transformations.
    147 """
    148 assert isinstance(gen_fn, GenerativeFunction)
--> 149 return lambda *args: _trace(gen_fn, addr, *args)

File ~/code/python/genjax/src/genjax/_src/generative_functions/static/static_transforms.py:129, in _trace(gen_fn, addr, *args)
    127 static_check_address_type(addr)
    128 addr = tree_map_const(addr)
--> 129 return initial_style_bind(trace_p)(_abstract_gen_fn_call)(
    130     gen_fn,
    131     addr,
    132     *args,
    133 )

File ~/code/python/genjax/src/genjax/_src/core/interpreters/forward.py:123, in initial_style_bind.<locals>.bind.<locals>.wrapped(*args, **kwargs)
    121 def wrapped(*args, **kwargs):
    122     """Runs a function and binds it to a call primitive."""
--> 123     _jaxpr, (flat_args, in_tree, out_tree) = stage(f)(*args, **kwargs)
    124     outs = prim.bind(
    125         *it.chain(_jaxpr.literals, flat_args),
    126         _jaxpr=_jaxpr.jaxpr,
   (...)
    130         **params,
    131     )
    132     return tree_util.tree_unflatten(out_tree(), outs)

File ~/code/python/genjax/src/genjax/_src/core/interpreters/staging.py:76, in stage.<locals>.wrapped(*args, **kwargs)
     74 flat_fun, out_tree = api_util.flatten_fun_nokwargs(fun, in_tree)
     75 flat_avals = safe_map(get_shaped_aval, flat_args)
---> 76 typed_jaxpr = cached_stage_dynamic(flat_fun, tuple(flat_avals))
     77 return typed_jaxpr, (flat_args, in_tree, out_tree)

    [... skipping hidden 1 frame]

File ~/code/python/genjax/src/genjax/_src/core/interpreters/staging.py:63, in cached_stage_dynamic(flat_fun, in_avals)
     61 @lu.cache
     62 def cached_stage_dynamic(flat_fun, in_avals):
---> 63     jaxpr, _, consts = pe.trace_to_jaxpr_dynamic(flat_fun, in_avals)
     64     typed_jaxpr = jax_core.ClosedJaxpr(jaxpr, consts)
     65     return typed_jaxpr

    [... skipping hidden 4 frame]

File ~/code/python/genjax/src/genjax/_src/generative_functions/static/static_transforms.py:118, in _abstract_gen_fn_call(gen_fn, _, *args)
    117 def _abstract_gen_fn_call(gen_fn, _, *args):
--> 118     return gen_fn.__abstract_call__(*args)

AttributeError: 'GaussianProcess' object has no attribute '__abstract_call__'
========================================================== short test summary info ==========================================================
FAILED notebooks/concepts/combinators/intro_to_combinators.ipynb::
FAILED notebooks/advanced/gensp/gensp.ipynb::
FAILED notebooks/concepts/gradient_interfaces/gradient_interfaces.ipynb::
FAILED notebooks/intermediate/gaussian_processes/gaussian_processes.ipynb::
======================================================= 4 failed, 8 passed in 28.24s ========================================================
```

File: pyproject.toml [hunk 4]
```diff
@@ -116,9 +117,7 @@ exclude = [
     "build",
     "dist",
     "node_modules",
-    "venv",
-    # TODO: Remove this exclusion once we fix the bitrot in this particular notebook.
-    "notebooks/internals/impl_builtin_language/impl_builtin_language.ipynb"
+    "venv"
```
Comment (littleredcomputer) [inline pyproject.toml:120]: feel free to leave trailing commas in python contexts (not necessary to merge)

**Review by littleredcomputer — APPROVED** [review]

**Pull request merged**